model/src/main/kotlin/ScanRecord.kt

```diff
@@ -53,7 +53,9 @@ data class ScanRecord(
 
         scanResults.forEach { (id, results) ->
             results.forEach { result ->
-                collectedIssues.getOrPut(id) { mutableSetOf() } += result.summary.issues
+                if (result.summary.issues.isNotEmpty()) {
+                    collectedIssues.getOrPut(id) { mutableSetOf() } += result.summary.issues
+                }
```

Comment:
Only semi-related: We should probably add a new test somewhere (but I don't know yet where) that verifies that the values returned by `OrtResult.collectIssues()` are never empty. But that's out of the scope of this PR.

Reply:
> but I don't know yet where

Probably in `OrtResultTest`.

Reply:
Hm, I just noticed that `AdvisorRecordTest` expects `collectIssues()` to return empty lists. 
So, this seems intended in the advisor.

Reply:
> So, this seems intended in the advisor.

I believe I had discussed this with @oheger-bosch once; IIRC he was in favor of keeping the empty lists as the caller can then rely on every `id` to be present as a key, but I was in favor of removing it, to more quickly be able to see which `id`s don't have issues.

Also, I already checked that for the `AnalyerResult` also only non-empty issues are added, so I'd prefer to align on that. Any more thoughts about this @oheger-bosch?

Reply:
@sschuberth @oheger-bosch, I've updated the test to the new behavior.

Reply:
@oheger-bosch will only be back on Monday to chime in to this discussion.

Reply:
> > So, this seems intended in the advisor.
> 
> I believe I had discussed this with @oheger-bosch once; IIRC he was in favor of keeping the empty lists as the caller can then rely on every `id` to be present as a key, but I was in favor of removing it, to more quickly be able to see which `id`s don't have issues.
> 
> Also, I already checked that for the `AnalyerResult` also only non-empty issues are added, so I'd prefer to align on that. Any more thoughts about this @oheger-bosch?

I would also prefer that, because the benefit of being able to use `collectIssues().getValue(id).isEmpty()` instead of `collectIssues()[id]?.isEmpty() == true` is minimal. And the first relies on the context knowledge that the map really contains keys for all packages which is already wrong if `skipExcluded` is set to `true` and there are excluded packages.

Reply:
Or even `id !in collectIssues().keys()`.

Reply:
These are good arguments, and I have no strong feelings about it. So, no objections from my side.